hazelcast/src/main/java/com/hazelcast/security/impl/function/SecuredFunctions.java

```diff
@@ -274,24 +275,29 @@ public List<Permission> permissions() {
 
     public static <T> ProcessorSupplier readJdbcProcessorFn(
             String connectionUrl,
-            FunctionEx<ProcessorSupplier.Context, ? extends DataSource> newDataSourceFn,
+            FunctionEx<ProcessorSupplier.Context, ? extends DataStoreHolder<DataSource>> newDataSourceFn,
             ToResultSetFunction resultSetFn,
             FunctionEx<? super ResultSet, ? extends T> mapOutputFn
     ) {
         return new ProcessorSupplier() {
 
-            private DataSource dataSource;
+            private DataStoreHolder<DataSource> dataSource;
 
             @Override
             public void init(@Nonnull ProcessorSupplier.Context context) {
                 dataSource = newDataSourceFn.apply(context);
             }
 
+            @Override
+            public void close(Throwable error) throws Exception {
```

Comment:
seems we don't need this newly introduced `close` method anymore since data-source is always `nonClosing`

Reply:
here we can pass a non-shared datastore that needs to be closed:

https://github.com/hazelcast/hazelcast/blob/d6f389236ca5e380e01f6cdbc6e23be60a8b9c59/hazelcast/src/main/java/com/hazelcast/jet/core/processor/SourceProcessors.java#L460

Reply:
Makes sense, my point is, i don't like an extra class in api and if it is possible i like to remove it, that extra class is DataStoreHolder in this case. Instead of it, when there is a shared resource, we can return a wrapped non-closable `DataSource` and unwrap it during node shut-down to close it. 

But in its current form, this pr is also doing its job. Thanks.